CONTRIBUTING.md

```diff
@@ -0,0 +1,120 @@
+<!-- 
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+# Contributing to Bolt
+
+Welcome! Thank you for your interest in the Bolt project. Before starting to contribute, please take a moment to review the guidelines outlined below.
+
+Bolt is a high-performance execution engine designed to replace and surpass existing standards. We value **performance, correctness, and code quality** above all else.
+
+## 🤝 Ways to Contribute
+
+Contributions are not just about code. Here are a few ways you can help:
+
+*   **Reporting Bugs**: If you find a crash, wrong result, or build failure, please file a [Bug Report](.github/ISSUE_TEMPLATE/01_bug_report.yml).
+*   **Performance Tuning**: Bolt is all about speed. If you identify a bottleneck, file a [Performance Issue](.github/ISSUE_TEMPLATE/06_performance_issue.yml) or submit an optimization PR.
+*   **Documentation**: Improve our docs, fix typos, or translate guides.
+*   **Answering Questions**: Help other users in [Discussions](https://github.com/bytedance/bolt/discussions).
+
+## 🛠 Development Setup
+
+Bolt uses modern C++ tooling. Please ensure your environment meets the requirements.
+
+### Prerequisites
+*   **OS**: Linux (Ubuntu 20.04+, CentOS 7+). macOS is currently experimental.
```

Comment:
Do we even want to claim it can be compiled on macos? I don't think it works currently. Additionally, CentOS 7 is EOL as of June 30, 2024. 

The only open distribution of CentOS now is CentOS Stream which I don't think we have built or tested on before. It might be just best to claim current supported debian LTS releases (bullseye, bookworm, trixie)